compiler/src/dotty/tools/dotc/semanticdb/SyntheticsExtractor.scala

```diff
@@ -24,7 +25,9 @@ class SyntheticsExtractor:
             tree.args.forall(arg =>
               arg.symbol.isOneOf(GivenOrImplicit) &&
               arg.span.isSynthetic
-            ) =>
+            ) &&
+            // exclude synthesized method parseArgument by @main
+            tree.symbol != defn.CLP_parseArgument =>
```

Comment:
this isn't enough to cover all the `@main` boilerplate, e.g. a main method with varargs will call the `parseRemainingArguments` and insert implicit args e.g. :
```scala
@main def readInts(ints: Int*): Unit = println(ints.mkString(","))
```
this will produce the synthetic:
```

Synthetics:
[6:0..6:0): => *(given_FromString_Int)
```

Reply:
A way to avoid this and more is to add a case to `ExtractSemanticDB` for `case tree: TypeDef` where we do not traverse the children if the symbol is `Invisible` -

```scala
case tree: TypeDef =>
  if !tree.symbol.is(Invisible) then
    traverseChildren(tree)
```
this seems to only affect classes generated by `@main` so far - will need @tgodzik to verify this is ok as it will remove all occurrences

Reply:
alternatively adding a case in the `Template` traversal where we avoid the body if the class symbol is `Invisible`